### ISSUE TITLE
The universal selector (*) works with :has() pseudo-class?

### ISSUE THREAD

**Issue by yamahige** [vivliostyle/vivliostyle.js#1341]
This css may cause an error "TypeError:` Cannot read properties of null (reading 'children')":
```
*:has(+ h2)::after {
  content: "[section boundary]";
  color: crimson;
  display: block;
}
```

This css works:
```
:is(p, div):has(+ h2)::after {
  content: "[section boundary]";
  color: crimson;
  display: block;
}
```

sample:
https://gist.githubusercontent.com/yamahige/9960e05fdf4bf611771ecac83387bf9d/raw/285ba3e86c0127a29fe9002b671ec27d065dea77/universal_selector_w_has

Test with Vivliostyle Viewer :
https://vivliostyle.org/viewer/#src=https://gist.githubusercontent.com/yamahige/9960e05fdf4bf611771ecac83387bf9d/raw/285ba3e86c0127a29fe9002b671ec27d065dea77/universal_selector_w_has

Result screenshot (version: 2.29.0):
![TypeError](https://github.com/vivliostyle/vivliostyle.js/assets/28525542/052a00c0-1458-424e-be1c-9591e80ba807)

expected:
![expected](https://github.com/vivliostyle/vivliostyle.js/assets/28525542/4bad5251-3db0-4556-ad32-88a21c0d6078)

**Comment by MurakamiShinyu** [comment]
@yamahige Thanks for reporting the issue.

The problem seems to be about the `:has()` pseudo class with sibling combinator (`+` or `~`) on the root element, which the universal selector `*` also selects. Examples:

```css
html:has(+ foo) {}
html:has(~ foo) {}
:root:has(+ foo) {}
:root:has(~ foo) {}
*:has(+ foo) {}
*:has(~ foo) {}
:has(+ foo) {}
:has(~ foo) {}
```

**Comment by MurakamiShinyu** [comment]
Fixed in #1342.

Test: https://vivliostyle-git-fix-issue1341-vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/yamahige/9960e05fdf4bf611771ecac83387bf9d/raw/285ba3e86c0127a29fe9002b671ec27d065dea77/universal_selector_w_has

**Comment by yamahige** [comment]
> about the :has() pseudo class with sibling combinator (+ or ~) on the root element,

I see.

> Fixed in https://github.com/vivliostyle/vivliostyle.js/pull/1342.

Thanks
